iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,342 @@
+    tooLongName.insert(0, 1, '/');
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(tooLongName); },
+                 "Application name has more than 100 characters, including null termination!");
+}
+
+
+TEST_F(PoshRuntime_test, MaxAppNameLength)
+{
+    std::string maxValidName(99, 's');
+    maxValidName.insert(0, 1, '/');
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(maxValidName); });
+}
+
+
+TEST_F(PoshRuntime_test, NoAppName)
+{
+    const std::string invalidAppName("");
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(invalidAppName); },
+                 "Cannot initialize runtime. Application name must not be empty!");
+}
+
+
+TEST_F(PoshRuntime_test, NoLeadingSlashAppName)
+{
+    const std::string invalidAppName = "invalidname";
+
+    EXPECT_DEATH(
+        { PoshRuntime::getInstance(invalidAppName); },
+        "Cannot initialize runtime. Application name invalidname does not have the required leading slash '/'");
+}
+
+
+// since getInstance is a singleton and test class creates instance of Poshruntime,
+// when getInstance() is called without parameterx it reuturns extisting instance
+// To be able to test this, it needs to be the very first call to getInstance but since,
+// we have multiple tests in this binary its not possible here to test
+TEST_F(PoshRuntime_test, DISABLED_AppNameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceNameIsSuccessful)
+{
+    const std::string appname = "/app";
+
+    auto& sut = PoshRuntime::getInstance(appname);
+
+    EXPECT_EQ(sut.getInstanceName(), appname);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplicationIsSuccessful)
+{
+    uint32_t uniqueIdCounter = iox::popo::BasePortData::s_uniqueIdCounter;
+
+    const auto applicationPortData = m_runtime->getMiddlewareApplication();
+
+    ASSERT_THAT(applicationPortData, Ne(nullptr));
+    EXPECT_THAT(m_runtimeName, applicationPortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), applicationPortData->m_serviceDescription);
+    EXPECT_EQ(false, applicationPortData->m_toBeDestroyed);
+    EXPECT_EQ(uniqueIdCounter, applicationPortData->m_uniqueId);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ApplicationlistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    // i=1 because there is already an active runtime in test fixture class which acquired an application port
+    for (auto i = 1u; i < iox::MAX_PROCESS_NUMBER; ++i)
+    {
+        auto appPort = m_runtime->getMiddlewareApplication();
+        EXPECT_NE(nullptr, appPort);
+    }
+    auto appPort = m_runtime->getMiddlewareApplication();
```

Comment:
please add a check that the error handler was not called before you force the overflow. With the current checks it is possible that the error handler is already called with any invocation of getMiddlewareApplication and we wouldn't notice it

Reply:
added